### PR TITLE
Fix Gadget API-Do not echo stuff to wikipedia

### PR DESCRIPTION
**Pull request by GlazerMann**
Fixes Gadget API.

Bot sends this when processes page with simply "MyText":
[15:36:43] Processing page ...
{"expandedtext":"MyText","editsummary":"\/* Testing *\/  | [[WP:UCB|Assisted by Citation bot]]"}
Leads to error on wiki side because of processing page line is not valid json:
SyntaxError: JSON.parse: expected ',' or ']' after array element at line 2 column 4 of the JSON data

I was confused about cause.  Turns out that HTML had nothing to do with it.

This turns on the eating of the buffer sooner.

### CODE REVIEW

**Comment by GlazerMann** [comment]
The existing code does not work because defines do not get passed in.

**Comment by GlazerMann** [comment]
I think this is now done.

**Comment by ms609** [comment]
This brings to my attention the lack of any test cases for `gadgetapi.php` ... at a minimum, is it possible to create a case (probably in a new file gadgetapiTest.php in the testcases folder) that checks that the output of the page is valid JSON in a couple of instances?

**Comment by GlazerMann** [comment]
>  is it possible to create a case that checks that the output of the page is valid JSON in a couple of instances.

I do not think that this is straight forward.  We would need to:
1.  setup PHP's webserver to run the gadgetapi.php page.  See:   http://symfony.com/doc/current/setup/built_in_web_server.html
2. query this webserver with an http request and save the output
3. parse this data as json and verify it

**Comment by GlazerMann** [comment]
Getting the white space right was a pain in the JSON.  We now have a "test" of sorts.

**Comment by GlazerMann** [comment]
I am done with this.  Please commit or comment.

File: tests/phpunit/TemplateTest.php [hunk 1]
```diff
@@ -564,6 +564,30 @@ public function testConvertJournalToBook() {
        $this->assertEquals('cite book', $expanded->wikiname());
    }
     
+   public function testGadgetAPI() {
```
Comment (ms609) [inline tests/phpunit/TemplateTest.php]: What we need to test, if anything, is that nothing is echoed when gadgetapi.php is run (except the final output).  As you say, the functionality itself is tested in TemplateTest.php and elsewhere.  It's not clear that this tests that.

**Pull request merged**